core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceService.scala

```diff
@@ -293,41 +308,62 @@ class MultiCloudWorkspaceService(override val ctx: RawlsRequestContext,
         "Created azure workspace " +
           s"[ workspaceId='${newWorkspace.workspaceId}'" +
           s", workspaceName='${newWorkspace.toWorkspaceName}'" +
-          //  s", cloneJobReportId='${cloneResult.getJobReport.getId}'" + WOR-697
+          s", containerCloneJobReportId='${containerCloneResult.getJobReport.getId}'" +
           s"]"
       )
 
-      // We can't specify the jobId to WSM at the time of writing and
-      // a 22-character base64url-encoded short-UUID is generated instead.
-      // Commented out until WOR-697
-//      jobId = WorkspaceManagerDAO.decodeShortUuid(cloneResult.getJobReport.getId).getOrElse {
-//        // If this happens we're in trouble. Our database needs a UUID but whatever WSM gave us
-//        // was not something we know how to convert into a UUID. I don't think we should delete the
-//        // workspace record but I'm not going to write an elaborate error handler to clean this up,
-//        // instead favouring PF-1269 as a better solution.
-//        val message =
-//          "Job report id returned by WSM when creating workspace clone was not a ShortUUID " +
-//            s"[ workspaceName='${newWorkspace.toWorkspaceName}'" +
-//            s", workspaceId='${newWorkspace.workspaceId}'" +
-//            s", jobId='${cloneResult.getJobReport.getId}' " +
-//            s"]"
-//
-//        logger.error(message)
-//        throw RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.InternalServerError, message))
-//      }
-//
-//      // hand off monitoring the clone job to the resource monitor
-//      _ <- WorkspaceManagerResourceMonitorRecordDao(dataSource).create(
-//        WorkspaceManagerResourceMonitorRecord.forCloneWorkspace(
-//          jobId,
-//          newWorkspace.workspaceIdAsUUID,
-//          parentContext.userInfo.userEmail
-//        )
-//      )
+      // hand off monitoring the clone job to the resource monitor
+      _ <- WorkspaceManagerResourceMonitorRecordDao(dataSource).create(
+        WorkspaceManagerResourceMonitorRecord.forCloneWorkspace(
+          UUID.fromString(containerCloneResult.getJobReport.getId),
+          newWorkspace.workspaceIdAsUUID,
+          parentContext.userInfo.userEmail
+        )
+      )
 
     } yield newWorkspace
   }
 
+  def cloneWorkspaceStorageContainer(sourceWorkspaceId: UUID,
+                                     destinationWorkspaceId: UUID,
+                                     ctx: RawlsRequestContext
+  ): Future[CloneControlledAzureStorageContainerResult] = {
+
+    val expectedContainerName = getStorageContainerName(sourceWorkspaceId)
+    val allContainers =
+      workspaceManagerDAO.enumerateStorageContainers(sourceWorkspaceId, 0, 200, ctx).getResources.asScala
```

Comment:
Choice of max 200 is arbitrary… I'm having trouble imagining a case where we would have more than that.

Reply:
maybe a good idea to note that inline with a comment? 
just to keep someone later from being afraid to change it because they assume there's some sort of significance to it

Reply:
Added comment.